src/main/java/com/denizenscript/denizencore/scripts/commands/file/FileReadCommand.java

```diff
@@ -52,55 +53,63 @@ public FileReadCommand() {
 
     public static void autoExecute(final ScriptEntry scriptEntry,
                                    @ArgPrefixed @ArgName("path") final String path) {
+        File file = canRead(path, scriptEntry);
+        if (file == null) {
+            return;
+        }
+        Runnable runme = () -> {
+            try {
+                FileInputStream stream = new FileInputStream(file);
+                byte[] data = stream.readAllBytes();
+                stream.close();
+                scriptEntry.saveObject("data", new BinaryTag(data));
+                scriptEntry.setFinished(true);
+            }
+            catch (Exception e) {
+                Debug.echoError(scriptEntry, e);
+                scriptEntry.setFinished(true);
+            }
+        };
+        if (scriptEntry.shouldWaitFor()) {
+            DenizenCore.runAsync(runme);
+        }
+        else {
+            runme.run();
+        }
+    }
+
+    public static File canRead(String path, ScriptEntry scriptEntry) {
         if (!CoreConfiguration.allowFileRead) {
-            Debug.echoError(scriptEntry, "FileRead disabled in Denizen/config.yml (refer to command documentation).");
+            Debug.echoError(scriptEntry, "FileRead disabled in configuration (refer to command documentation).");

```

Comment:
`in configuration`, is (A) not English and (B) not really clear - what configuration? If altering this to be compatible with non-bukkit impl (ie clientizen) there should probably be some CoreImpl string that fills this spot with something coherent local to the impl